### ISSUE TITLE
Enhancement: Print version of typescript-espree in version warning

### ISSUE THREAD

**Issue by Mitsunee** [typescript-eslint/typescript-eslint#10272]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

N/A

### Description

I am yet again sitting here on a Monday with my favorite warning:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=4.7.4 <5.6.0

YOUR TYPESCRIPT VERSION: 5.6.3

Please only submit bug reports when using the officially supported version.

=============
```

It would be very useful if this warning could actually print the version(s) of typescript-eslint packages involved (especially typescript-espree) to help with debugging. I have manually crawled through my lockfile 3 times and am pretty sure I should be on version 8.12.2, but have no way to confirm this.

### Fail

```TypeScript
/
```

### Pass

```TypeScript
/
```

**Comment by kirkwaiblinger** [comment]
I'm +1. I've run into sticky situations myself where this would have been useful (involving multiple installations of typescript-eslint and a single script with separate invocations of eslint for different packages)

**Comment by JoshuaKGoldberg** [comment]
I'm also +1. I could have sworn I'd seen (or maybe filed?) a similar issue, but can't find it now...
